test/e2e/traceflow_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+)
+
+func TestTraceflow(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	nodeName := workerNodeName(1)
+
+	podName0 := randName("test-pod-")
+	t.Logf("Creating Pod '%s' on '%s'", podName0, nodeName)
+	if err := data.createBusyboxPodOnNode(podName0, nodeName); err != nil {
+		t.Fatalf("Error when creating Pod '%s': %v", podName0, err)
+	}
+	defer deletePodWrapper(t, data, podName0)
+	if _, err := data.podWaitForIP(defaultTimeout, podName0, testNamespace); err != nil {
+		t.Errorf("Error when waiting for IP for Pod '%s': %v", podName0, err)
+	}
+
+	podName1 := randName("test-pod-")
+	t.Logf("Creating Pod '%s' on '%s'", podName1, nodeName)
+	if err := data.createBusyboxPodOnNode(podName1, nodeName); err != nil {
+		t.Fatalf("Error when creating Pod '%s': %v", podName1, err)
+	}
+	defer deletePodWrapper(t, data, podName1)
+	if _, err := data.podWaitForIP(defaultTimeout, podName1, testNamespace); err != nil {
+		t.Errorf("Error when waiting for IP for Pod '%s': %v", podName1, err)
+	}
+
+	tf := &v1alpha1.Traceflow{
+		Spec: v1alpha1.TraceflowSpec{
+			Source: v1alpha1.Source{
+				Namespace: testNamespace,
+				Pod:       podName0,
+			},
+			Destination: v1alpha1.Destination{
+				Namespace: testNamespace,
+				Pod:       podName1,
+			},
+			Packet: v1alpha1.Packet{
+				IPHeader: v1alpha1.IPHeader{
+					Protocol: 6,
+				},
+				TransportHeader: v1alpha1.TransportHeader{
+					TCP: &v1alpha1.TCPHeader{
+						DstPort: 80,
+					},
+				},
+			},
+		},
+	}
+
+	localConfig := rest.CopyConfig(data.kubeConfig)
+	client, err := versioned.NewForConfig(localConfig)
+	require.NoError(t, err)
+
+	var allowAllIngress *networkingv1.NetworkPolicy
+	if allowAllIngress, err = data.createNetworkPolicyAllowAllIngress("test-networkpolicy-allow-all-ingress"); err != nil {
```

Comment:
I think it is better to create a NetworkPolicy to deny the traffic, like a Policy to allow egress and another one to disallow ingress. But the current is a good start.

Reply:
Update with 2 NetworkPolicies: allow egress all and deny ingress with label.